packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -587,3 +587,403 @@ class MultiTapGestureRecognizer extends GestureRecognizer {
+
+  /// A pointer has contacted the screen at a particular location\, which might
+  /// be the start of a serial tap.
+  ///
+  /// If this recognizer loses the arena before the serial tap is completed
+  /// (either because the gesture does not end up being a tap or because another
+  /// recognizer such a [TapGestureRecognizer] or [DoubleTapGestureRecognizer]
+  /// wins outright), [onSerialTapCancel] is called next. Otherwise,
+  /// [onSerialTapUp] is called next.
+  ///
+  /// The [SerialTapDownDetails.count] that is passed to this callback
+  /// specifies the series tap count.
+  GestureSerialTapDownCallback? onSerialTapDown;
+
+  /// A pointer that previously triggered [onSerialTapDown] will not end up
+  /// triggering the corresponding [onSerialTapUp].
+  ///
+  /// If the user completes the serial tap, [onSerialTapUp] is called instead.
+  ///
+  /// The [SerialTapCancelDetails.count] that is passed to this callback will
+  /// match the [SerialTapDownDetails.count] that was passed to the
+  /// [onSerialTapDown] callback.
+  GestureSerialTapCancelCallback? onSerialTapCancel;
+
+  /// A pointer has stopped contacting the screen at a particular location,
+  /// representing a serial tap.
+  ///
+  /// If the user didn't complete the tap, or if another recognizer won the
+  /// arena before this recognizer handled the completed tap, then
+  /// [onSerialTapCancel] is called instead.
+  ///
+  /// The [SerialTapUpDetails.count] that is passed to this callback
+  /// specifies the series tap count.
+  ///
+  /// Because this callback fires for every completed serial tap, this callback
+  /// may fire even when this recognizer doesn't win the arena. For example, if
+  /// if it is competing with [DoubleTapGestureRecognizer], both recognizers
+  /// will remain in the arena after the first tap, and [onSerialTapUp] will
+  /// fire with a [SerialTapUpDetails.count] of 1. Then, after the second tap,
+  /// the double-tap recognizer will win the arena.
+  GestureSerialTapUpCallback? onSerialTapUp;
+
+  Timer? _serialTapTimer;
+  final List<_TapTracker> _completedTaps = <_TapTracker>[];
+  _TapTracker? _pendingTap;
+
+  /// Indicates whether this recognizer is currently tracking a pointer that's
+  /// in contact with the screen.
+  ///
+  /// If this is true, it implies that [onSerialTapDown] has fired, but neither
+  /// [onSerialTapCancel] nor [onSerialTapUp] have yet fired.
+  bool get isTrackingPointer => _pendingTap != null;
+
+  @override
+  bool isPointerAllowed(PointerDownEvent event) {
+    if (_completedTaps.isEmpty) {
+      if (onSerialTapDown == null && onSerialTapCancel == null && onSerialTapUp == null) {
+        return false;
+      }
+    }
+    return super.isPointerAllowed(event);
+  }
+
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    if (_completedTaps.isNotEmpty && !_representsSameSeries(_completedTaps.last, event)) {
+      _reset();
+    } else if (_pendingTap != null) {
```

Comment:
The sequence down-1, down-2, up-1, up-2 will cause the events Down(count: 1), Cancel(count: 1), Down(count: 1), Up(count: 1).

Maybe document on the class how it handles interleaving Taps since this is different from the regular TapRecognizer as well as the MultiTap recognizer...

Reply:
Done (in the class docs).  I also added a test for this case.